src/Http/Authentication.Core/src/AuthenticationHandlerProvider.cs

```diff
@@ -49,9 +49,22 @@ public AuthenticationHandlerProvider(IAuthenticationSchemeProvider schemes)
             {
                 return null;
             }
-            var handler = (context.RequestServices.GetService(scheme.HandlerType) ??
-                ActivatorUtilities.CreateInstance(context.RequestServices, scheme.HandlerType))
-                as IAuthenticationHandler;
+
+            var handler = (IAuthenticationHandler?) context.RequestServices.GetService(scheme.HandlerType);
+            if (handler == null)
+            {
+                if (scheme is AuthenticationSchemeWithFactory schemeWithFactory)
```

Comment:
Shouldn't it be possible to get rid of `AuthenticationSchemeWithFactory` and just create the object factory inline here? The only reason not to do that would be to avoid creating an object factory for every scheme every time `GetHandlerAsync` is called, but since the handler is stored in `_handlerMap` there shouldn't be multiple calls for the same scheme anyways.

Reply:
AuthenticationHandlerProvider is scoped so it's re-created per-request anyways. That makes _handlerMap short lived.